beta/src/hooks/usePendingRoute.ts

```diff
@@ -0,0 +1,33 @@
+import {useRouter} from 'next/router';
+import React from 'react';
+
+const usePendingRoute = () => {
+  const {events} = useRouter();
+  const [pendingRoute, setPendingRoute] = React.useState<string | null>(null);
+  const currentRoute = React.useRef<string | null>(null);
+  React.useEffect(() => {
+    let routeTransitionTimer: any = null;
+
+    const handleRouteChange = (url: string) => {
+      routeTransitionTimer = setTimeout(() => {
+        if (currentRoute.current === url) return;
+        currentRoute.current = url;
+        setPendingRoute(url);
+      }, 100);
+    };
+    const handleRouteChangeComplete = () => {
+      setPendingRoute(null);
+      clearTimeout(routeTransitionTimer);
+    };
+    events.on('routeChangeStart', handleRouteChange);
+    events.on('routeChangeComplete', handleRouteChangeComplete);
```

Comment:
I think we need to handle routeChangeError (abort) as well, in case the user clicks another link or the back button, while the route is still loading. Or clear the timeout before setting up a new one in line 12?